website/docs/pipelines/triggers.md

```diff
@@ -0,0 +1,145 @@
+---
+sidebar_position: 1
+sidebar_label: Pipeline Triggers
+---
+
+# The role of triggers in pipelines
+
+Triggers in a CI/CD pipeline are more than just automated switches; they embody
+the principles of flow and harmony within software development. By defining when
+and how processes like building, testing, or deploying should happen, triggers
+shape the rhythm of the pipeline. If set with precision, they ensure that every
+change is seamlessly integrated, tested, and delivered, maintaining the
+integrity of the repository. However, if misconfigured, they can lead to
+inconsistencies, breaking the cadence of development and introducing chaos into
+what should be a continuous, smooth evolution. In essence, triggers dictate the
+pipeline's scope, serving as the silent guardians of coherence and reliability.
+
+This document provides a guide on how triggers should be set when using the
+GitHub Actions template provided by this repository.
+
+:::note
+
+Every pipeline should have the `workflow_dispatch` trigger. In this way we will
+always be able to launch it from the UI, which may be necessary in moments when
+it is necessary to exit the ordinary flow (i.e. incidents).
+
+```yaml
+on:
+  workflow_dispatch:
+```
+
+:::
+
+## Continuous Integration on GitHub Actions
+
+Continuous integration refers to the build, validation and testing of the
+software. Generally, these tasks should be launched when a Pull Request is
+opened against the main branch, to ensure that high code quality is maintained.
+
+Then, a pipeline in charge of validate code (unit test, linter, Terraform Plans,
+etc.) should run when:
+
+- a PR is opened
+- a commit is added to an already opened PR
+- a previously closed PR is reopened
+- a draft PR is marked as ready for review
+
+To achieve these settings, use the following:
+
+```yaml
+on:
+  pull_request:
+    types:
+      - opened # Trigger on new PRs
+      - synchronize # Trigger when commits are added to an open PR
+      - reopened # Trigger when a closed PR is reopened
+      - ready_for_review # Trigger when a draft PR is marked as ready
+```
+
+At the end, add the trigger paths which would make sense to you. These generally
+are:
+
+- Terraform:
+  - the path to the specific Terraform configuration, including both environment
+    (i.e. `prod`) and modules (`_modules`) folders
+- the GitHub workflow file you are working on, including also the CD file
+- apps:
+  - the workspace of a given application
+
+Here's an example of a good setup for Terraform CI workflows:
+
+```yaml
+on:
+  pull_request:
+    types:
+      - ...
+    paths:
+      - "src/core/prod/**"
```

Comment:
why core/prod? shouldn't be infra/resources/** and infra/modules/** ?

Reply:
I took the example from `io-infra`, but I agree with you as your example is more representative